subprojects/core/src/main/java/org/gradle/api/internal/tasks/DefaultTaskContainer.java

```diff
@@ -646,7 +647,7 @@ public Action<? super Task> whenObjectRemoved(Action<? super Task> action) {
     }
 
     @Override
-    public void whenObjectRemoved(Closure action) {
+    public void whenObjectRemoved(@DelegatesTo(Task.class) Closure action) {
```

Comment:
Does it make sense to add the `@DelegatesTo` annotation to the implementation, even it is already on the interface? Is the annotation on the interface enough?

Reply:
Annotation on the interface should be enough. I will sort that out.